client/src/pages/Login.js

```diff
@@ -200,36 +204,86 @@ const Login = ({ isLoginForm }) => {
     setConfirmPassword(e.target.value);
   };
 
+  const handleErrorMessage = (type) => {
+    if(type === "email"){
+      const email = document.getElementById("emailID");
```

Comment:
I see what you're doing here but using DOM selectors/methods to change state is not the "React" way of working with the DOM. This is because React uses the virtual DOM and a diffing algorithm to track changes of the DOM. When you're changing the innerText and the style color on lines 211 and 212 - this will confuse React and potentially cause performance issues. 

I think the "React" way of approaching this is to use booleans within the JSX to toggle classNames/elements to show/hide error messages. 

@robinv85 @snowiesuet What do you think?

Reply:
I think this has to be refactored anyway because I've added the react-hook-form library on my branch for the error messages and that one handles most of the validation logic already ...

Reply:
and that library has a hook `useForm` that returns an `errors` object and different ways to handle validation etc. ... so probably better to merge my branch and then just finish the details with work from this branch for showing the actual validation messages

Reply:
Also one problem with these form inputs from Ant design is that they don't expose a prop to get the ref, other similar UI libraries do this with `InnerRef` that's also why I had to use the `Control` component to handle some of the input values...